### PR TITLE
glance: Update glance config templates for Mitaka

### PR DESCRIPTION
**Pull request by toabctl**
Update the configuration templates for glance

### CODE REVIEW

**Comment by sayalilunkad** [comment]
Looks good, except maybe you want to remove the `use_syslog` as well since it is deprecated?

**Comment by toabctl** [comment]
> Looks good, except maybe you want to remove the use_syslog as well since it is deprecated?

It's not deprecated. See https://git.openstack.org/cgit/openstack/oslo.log/tree/oslo_log/_options.py?h=master#n87 .

**Comment by toabctl** [comment]
for the record: the option ```use_syslog``` is not deprecated. The release notes are wrong

**Comment by sayalilunkad** [comment]
@toabctl yup, thanks for confirming that. +1

File: chef/cookbooks/glance/templates/default/glance-api.conf.erb [hunk 1]
```diff
@@ -266,39 +257,80 @@ registry_port = <%= @registry_bind_port %>
...
+# Reason: This option was considered harmful and has been deprecated
+# in M release. It will be removed in O release. For more information
+# read OSSN-0060. Related functionality with uploading big images has
+# been implemented with Keystone trusts support.
 #auth_region = <None>
 
 # The protocol to use for communication with the registry server.
 # Either http or https. (string value)
 #registry_client_protocol = http
-registry_client_protocol = http
```
Comment (rhafer) [inline chef/cookbooks/glance/templates/default/glance-api.conf.erb]: I know it's nothing new, but shouldn't we set this to ```https``` when ssl is enabled? Would of course also mean to enable SSL support for the glance registry service. Is that something we should track in the backlog?

File: chef/cookbooks/glance/templates/default/glance-api.conf.erb [hunk 2]
```diff
@@ -226,15 +216,16 @@ key_file = <%= node[:glance][:ssl][:keyfile] if node[:glance][:api][:protocol] =
 #image_cache_driver = sqlite
 
 # The upper limit (the maximum size of accumulated cache in bytes)
-# beyond which pruner, if running, starts cleaning the images cache.
-# (integer value)
+# beyond which the cache pruner, if running, starts cleaning the image
+# cache. (integer value)
 #image_cache_max_size = 10737418240
```
Comment (rhafer) [inline chef/cookbooks/glance/templates/default/glance-api.conf.erb]: I think this needs to be ```<%= node[:glance][:image_cache_max_size] %>``` (the same value as in glance-cache.conf), see http://docs.openstack.org/developer/glance/cache.html#configuration-options-for-the-image-cache

File: chef/cookbooks/glance/templates/default/glance-api.conf.erb [hunk 3]
```diff
@@ -226,15 +216,16 @@ key_file = <%= node[:glance][:ssl][:keyfile] if node[:glance][:api][:protocol] =
...
 
-# The amount of time to let an image remain in the cache without being
-# accessed. (integer value)
+# The amount of time to let an incomplete image remain in the cache,
+# before the cache cleaner, if running, will remove the incomplete
+# image. (integer value)
 #image_cache_stall_time = 86400
```
Comment (rhafer) [inline chef/cookbooks/glance/templates/default/glance-api.conf.erb]: This also needs to have the same value as in glance-cache.conf

Comment (toabctl) [inline chef/cookbooks/glance/templates/default/glance-api.conf.erb]: @rhafer No. The value is not used when running the glance-api service.

File: chef/cookbooks/glance/templates/default/glance-api.conf.erb (hunk 2 above)
Comment (toabctl) [inline chef/cookbooks/glance/templates/default/glance-api.conf.erb]: Done

File: chef/cookbooks/glance/templates/default/glance-api.conf.erb (hunk 3 above)
Comment (toabctl) [inline chef/cookbooks/glance/templates/default/glance-api.conf.erb]: Done

File: chef/cookbooks/glance/templates/default/glance-api.conf.erb (hunk 1 above)
Comment (toabctl) [inline chef/cookbooks/glance/templates/default/glance-api.conf.erb]: I created https://trello.com/c/nMBPFtws/1129-enable-https-for-glance-registry

**Comment by rhafer** [comment]
+1

**Comment by sayalilunkad** [comment]
Still +1 for me. Gating fail was unrelated so I rebuild it. Let's wait to see if it passes.

**Pull request merged**